numpy/ma/tests/test_core.py

```diff
@@ -1990,6 +2001,205 @@ def test_datafriendly_mul_arrays(self):
         assert_equal(a, [[1, 1], [3, 3]])
         assert_equal(a.mask, [[0, 1], [0, 1]])
 
+    def test_inplace_addition_scalar_type(self):
+        # Test of inplace additions
+        for t in self.othertypes:
+            with warnings.catch_warnings(record=True) as w:
+                warnings.filterwarnings("always")
+                (x, y, xm) = (_.astype(t) for _ in self.uint8data)
+                xm[2] = masked
+                x += t(1)
+                assert_equal(x, y + 1)
+                xm += t(1)
+                assert_equal(xm, y + 1)
+
+                assert_equal(len(w), 0, "Failed on type=" + repr(t) + ".")
+
+    def test_inplace_addition_array_type(self):
+        # Test of inplace additions
+        for t in self.othertypes:
+            with warnings.catch_warnings(record=True) as w:
+                warnings.filterwarnings("always")
+                (x, y, xm) = (_.astype(t) for _ in self.uint8data)
+                m = xm.mask
+                a = arange(10, dtype=t)
+                a[-1] = masked
+                x += a
+                xm += a
+                assert_equal(x, y + a)
+                assert_equal(xm, y + a)
+                assert_equal(xm.mask, mask_or(m, a.mask))
+
+                assert_equal(len(w), 0, "Failed on type=" + repr(t) + ".")
+
+    def test_inplace_subtraction_scalar_type(self):
+        # Test of inplace subtractions
+        for t in self.othertypes:
+            with warnings.catch_warnings(record=True) as w:
+                warnings.filterwarnings("always")
+                (x, y, xm) = (_.astype(t) for _ in self.uint8data)
+                x -= t(1)
+                assert_equal(x, y - 1)
+                xm -= t(1)
+                assert_equal(xm, y - 1)
+
+                assert_equal(len(w), 0, "Failed on type=" + repr(t) + ".")
+
+
+    def test_inplace_subtraction_array_type(self):
+        # Test of inplace subtractions
+        for t in self.othertypes:
+            with warnings.catch_warnings(record=True) as w:
+                warnings.filterwarnings("always")
+                (x, y, xm) = (_.astype(t) for _ in self.uint8data)
+                m = xm.mask
+                a = arange(10, dtype=t)
+                a[-1] = masked
+                x -= a
+                xm -= a
+                assert_equal(x, y - a)
+                assert_equal(xm, y - a)
+                assert_equal(xm.mask, mask_or(m, a.mask))
+
+                assert_equal(len(w), 0, "Failed on type=" + repr(t) + ".")
```

Comment:
I have never been clear on the proper way to handle string formatting. I remember seeing what you use here as standard when I first learned Python. Then, it was suggested that we use `.format` going forward, which to me seemed uglier and less readable. It appears the old strategy is technically deprecated even though it lives on ( https://www.python.org/dev/peps/pep-3101/ ) ( http://stackoverflow.com/questions/13451989/pythons-many-ways-of-string-formatting-are-the-older-ones-going-to-be-deprec ). I guess I figured you can't deprecate string concatenation and it is simple, but if you prefer the old formatting that's fine too.